lib/instance-validator.js

```diff
@@ -89,19 +90,27 @@ class InstanceValidator {
    * @private
    */
   validate() {
-    if (this.options.hooks) {
-      return this.modelInstance.constructor.runHooks('beforeValidate', this.modelInstance, this.options)
-        .then(() =>
-          this._validate().catch(error =>
-            this.modelInstance.constructor.runHooks('validationFailed', this.modelInstance, this.options, error).then(newError => {
-              throw newError || error;
-            })
-          )
-        )
-        .then(() => this.modelInstance.constructor.runHooks('afterValidate', this.modelInstance, this.options))
-        .return(this.modelInstance);
-    }
-    return this._validate();
+    return this.options.hooks ? this._validateAndRunHooks() : this._validate();
+  }
+
+  /**
+   * Invoke the Validation sequence and run hooks
+   *   - Before Validation Model Hooks
+   *   - Validation
+   *   - On validation success: After Validation Model Hooks
+   *   - On validation failure: Validation Failed Model Hooks
+   *
+   * @return {Promise}
+   * @private
+   */
+  _validateAndRunHooks() {
+    const runHooks = this.modelInstance.constructor.runHooks.bind(this.modelInstance.constructor);
+    return runHooks('beforeValidate', this.modelInstance, this.options)
+      .then(() => this._validate())
```

Comment:
I was just looking at the diff and realized these are equivalent. I'm catching beforeValidate failures too now